aws/cloudwatch_dashboard_test.go

```diff
@@ -0,0 +1,164 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListCloudWatchDashboards(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+	svc := cloudwatch.New(session)
+
+	cwdbName := createCloudWatchDashboard(t, svc, region)
+	defer deleteCloudWatchDashboard(t, svc, cwdbName, true)
+
+	cwdbNames, err := getAllCloudWatchDashboards(session, time.Now(), config.Config{})
+	require.NoError(t, err)
+	assert.Contains(t, aws.StringValueSlice(cwdbNames), aws.StringValue(cwdbName))
+}
+
+func TestTimeFilterExclusionNewlyCreatedCloudWatchDashboard(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+	svc := cloudwatch.New(session)
+
+	cwdbName := createCloudWatchDashboard(t, svc, region)
+	defer deleteCloudWatchDashboard(t, svc, cwdbName, true)
+
+	// Assert CWDB is picked up without filters
```

Comment:
NIT: could we maybe swap this comment abbreviation out for the full name? Just to keep it easier to read

Reply:
Done! cdaafce633cc5c357544350150269852ec2b583d